pynicotine/gtkgui/userbrowse.py

```diff
@@ -489,8 +489,8 @@ def browse_queued_path(self):
         if not iterator:
             return
 
-        # Scroll to the requested file
-        self.file_list_view.select_row(iterator)
+        # Scroll to the requested file and grab focus to it
+        self.file_list_view.select_row(iterator, should_expand=True)
```

Comment:
This feels strange, since we don't have expanders in the file list.

Reply:
I know, this is my least liked element of this PR.

Functionally it works perfectly, but it feels like a hack.

Suggest perhaps renaming/reworking the method in treeview.py to seperate:
- `should_scroll`
- `should_expand`
- `should_focus`

I cannot think of a scenario where we want to expand but not focus, but I could be mistaken.

Reply:
I think you could just call grab_focus() when should_focus is True. There shouldn't be any situations where we don't want to focus the widget when scrolling to a row.

Reply:
No, we can't do that because it breaks the search next match behaviour.

Reply:
That's a shame. I'll have to take a closer look, since the select_row() method is becoming dangerously large.

Reply:
I reverted the changes to treeview.py in favour of grabbing the focus in browse_queued_path()